library/index.c

```diff
@@ -323,24 +323,25 @@ int eblob_index_blocks_fill(struct eblob_base_ctl *bctl)
 }
 
 
-static struct eblob_disk_control *eblob_find_on_disk(struct eblob_backend *b,
-		struct eblob_base_ctl *bctl, struct eblob_disk_control *dc,
+static int eblob_find_on_disk(struct eblob_backend *b,
```

Comment:
I have found 2 issues in eblob_find_on_disk: one problem exists in current code and one thing I forget to add into this PR.
First one is about this lines:
if (num > b->cfg.index_block_size)
num = b->cfg.index_block_size;
// then bsearch only in this block, which contains only small part of index file.

Second one that I've forgotten to add implementation of bsearch with pread on each binary search iteration. So it reads whole index file into memory every time in this PR.
Also binary searching causes random read pattern, therefore index file must be prefetched into page cache before bin search.